dbt/adapters/databricks/connections.py

```diff
@@ -850,3 +856,88 @@ def _log_dbsql_errors(exc: Exception) -> None:
         logger.debug(f"{type(exc)}: {exc}")
         for key, value in sorted(exc.context.items()):
             logger.debug(f"{key}: {value}")
+
+
+def _should_poll_refresh(sql: str) -> Tuple[bool, str]:
+    # if the command was to refresh a materialized view we need to poll
+    # the pipeline until the refresh is finished.
+    name = ""
+    refresh_search = re.search(r"refresh\s+materialized\s+view\s+([`\w.]+)", sql)
+    if not refresh_search:
+        refresh_search = re.search(r"create\s+or\s+refresh\s+streaming\s+table\s+([`\w.]+)", sql)
+
+    if refresh_search:
+        name = refresh_search.group(1).replace("`", "")
+
+    return refresh_search is not None, name
+
+
+def _get_table_view_pipeline_id(host: str, headers: dict, name: str) -> str:
+    url = f"https://{host}/api/2.1/unity-catalog/tables/{name}"
+    resp1 = requests.get(url, headers=headers)
+    if resp1.status_code != 200:
+        raise dbt.exceptions.DbtRuntimeError(
+            f"Error getting materialized view/streaming table info: {name}"
+        )
+
+    pipeline_id = resp1.json().get("pipeline_id", "")
+    if not pipeline_id:
+        raise dbt.exceptions.DbtRuntimeError(
+            f"Error materialized view/streaming table {name} does not have a pipeline id"
+        )
+
+    return pipeline_id
+
+
+def _get_pipeline_state(host: str, headers: dict, pipeline_id: str) -> dict:
+    url = f"https://{host}/api/2.0/pipelines/{pipeline_id}"
+
+    response = requests.get(url, headers=headers)
+    if response.status_code != 200:
+        raise dbt.exceptions.DbtRuntimeError(f"Error getting pipeline info: {pipeline_id}")
+
+    return response.json()
+
+
+def _find_update(pipeline: dict, id: str = "") -> Optional[Dict]:
+    updates = pipeline.get("latest_updates", [])
+    if not updates:
+        raise dbt.exceptions.DbtRuntimeError(
+            f"No updates for pipeline: {pipeline.get('pipeline_id', '')}"
+        )
+
+    if not id:
+        return updates[0]
+
+    matches = [x for x in updates if x.get("update_id") == id]
+    if matches:
+        return matches[0]
+
+    return None
+
+
+def _get_update_error_msg(host: str, headers: dict, pipeline_id: str, update_id: str) -> str:
+    events_url = f"https://{host}/api/2.0/pipelines/{pipeline_id}/events"
+    response = requests.get(events_url, headers=headers)
+    if response.status_code != 200:
```

Comment:
Can we use `if response.ok` here? [docs](https://requests.readthedocs.io/en/latest/api/#requests.Response.ok)